pkg/apis/batch/v1alpha1/job.go

```diff
@@ -225,63 +225,83 @@ const (
 	Inqueue JobPhase = "Inqueue"
 )
 
-// JobState contains details for the current state of the job.
-type JobState struct {
-	// The phase of Job.
-	// +optional
-	Phase JobPhase `json:"phase,omitempty" protobuf:"bytes,1,opt,name=phase"`
-
-	// Unique, one-word, CamelCase reason for the phase's last transition.
-	// +optional
-	Reason string `json:"reason,omitempty" protobuf:"bytes,2,opt,name=reason"`
+type JobConditionType string
 
-	// Human-readable message indicating details about last transition.
-	// +optional
-	Message string `json:"message,omitempty" protobuf:"bytes,3,opt,name=message"`
+const (
+	// JobCreated means the job has been accepted by the system, but it hasn't
+	// reached the phase of running, this includes creating the job resource,
+	// waiting for pod resource and scheduling.
+	JobCreated JobConditionType = "JobCreated"
+	// JobRunning means at least 'MinAvailable' pods of this job are running.
+	JobRunning JobConditionType = "JobRunning"
+	// JobSucceed means the job has been successfully executed.
+	JobSucceed JobConditionType = "JobSucceed"
+	// JobFailed means the system failed to executed the job after a max retry.
+	JobFailed JobConditionType = "JobFailed"
+	// JobTerminated means the job has been terminated
+	JobTerminated JobConditionType = "JobTerminated"
+	// JobTerminated means the job has been suspended
+	JobAborted JobConditionType = "JobAborted"
+)
 
-	// Last time the condition transit from one phase to another.
-	// +optional
-	LastTransitionTime metav1.Time `json:"lastTransitionTime,omitempty" protobuf:"bytes,4,opt,name=lastTransitionTime"`
+// +k8s:deepcopy-gen=true
+// JobCondition describes the state of the job at a certain point.
+type JobCondition struct {
```

Comment:
This will make status more complicated, and if the condition like `JobSucceed` bacomes true, you need to set all other conditions false, which is not good

condition xxx, status: true/false
condition yyy, status: true/false
condition zzz, status: true/false

Reply:
We still have status/phase in JobStatus, JobConditions used to show some major changes during job lifecycle. Yeah, When jobSucceed the JobRunning condition would be false. But JobSucceed/Failed/Termiated/Aborted will not appear at the same time.

Reply:
I checked the pod condition, this make sense